goldilocks/src/lib.rs

```diff
@@ -262,6 +312,19 @@ impl Div for Goldilocks {
     }
 }
 
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Squares the base N number of times and multiplies the result by the tail value.
+#[inline(always)]
+fn exp_acc<const N: usize>(base: Goldilocks, tail: Goldilocks) -> Goldilocks {
+    let mut result = base;
+    for _ in 0..N {
+        result = result.square();
+    }
+    result * tail
```

Comment:
nit: there's an `exp_power_of_2` which almost does this

```suggestion
    base.exp_power_of_2(N) * tail
```

Reply:
thanks for the suggestion.